kqueen/blueprints/api/generic_views.py

```diff
@@ -140,15 +148,34 @@ class ListView(GenericView):
     action = 'list'
 
     def set_object(self, *args, **kwargs):
+        if request.args.get('all_namespaces'):
+            objects = []
+            for namespace in self.get_namespaces():
+                objects = objects + list(self.get_class().list(namespace, return_objects=True).values())
+            self.obj = objects
+            self.check_authorization()
+            return
+
         try:
-            namespace = current_identity.namespace
+            if request.args.get('namespace'):
+                namespace = request.args.get('namespace')
+            else:
+                namespace = current_identity.namespace
         except AttributeError:
             namespace = None
 
         self.obj = list(self.get_class().list(namespace, return_objects=True).values())
         self.check_authorization()
 
     def get_content(self, *args, **kwargs):
+        if request.args.get('all_namespaces'):
+            objs = []
+            for obj in self.obj:
+                namespace = obj._object_namespace
+                obj_dict = obj.get_dict(expand=True)
+                obj_dict['namespace'] = namespace
+                objs.append(obj_dict)
+            return objs
```

Comment:
We are returning:

* object for single namespace
* list of dicts (not objects) for all_namespaces

It this desired behavior? Why do we don't return list of object?*

* and add _namespace property to objects?

Reply:
I agree, that we could prefix the namespace attribute, so it does not look as default part of the schema for given resource, but I don't think whether its object or dict at this point, I think it gets serialized in the exactly same way either way before its returned.

Reply:
Thanks, waiting for fix.